src/algebra/big_operators/fin.lean

```diff
@@ -205,6 +205,31 @@ begin
     assoc_rw [hi, inv_mul_cancel_left] }
 end
 
+@[to_additive] lemma inv_partial_prod_mul_eq_contract_nth {G : Type*} [group G]
+  (g : fin (n + 1) → G) (a : fin (n + 1)) (x : fin n) :
+  (fin.partial_prod g (a.succ.succ_above x.cast_succ))⁻¹
+    * (fin.partial_prod g (a.succ_above x).cast_succ * g (a.succ_above x))
+    = a.contract_nth has_mul.mul g x :=
+begin
+  rcases lt_trichotomy (x : ℕ) a with (h|h|h),
```

Comment:
Is there a missing lemma that looks something like `contract_nth j op g (j.succ.succ_above  k) = g k`?

Reply:
I'll make
```
lemma contract_nth_apply_of_neq (j : fin (n + 1)) (op : α → α → α) (g : fin (n + 1) → α) 
  (k : fin n) (h : (j : ℕ) ≠ k) : 
 contract_nth j op g k = g (j.succ_above k) := ...
```

(though I don't think it reduces the length of the long proof)